### PR TITLE
Introduce patches

### PR DESCRIPTION
**Pull request by mleonowicz**

### CODE REVIEW

File: src/fontr/fontr/classifier.py [hunk 1]
```diff
@@ -106,13 +106,15 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0):
         return nn.functional.softmax(self.forward(batch), dim=1)
 
     def base_step(self, batch, batch_idx: int, step_name: str):
-        # TODO somehow we do not get labels heer
```
Comment (madziejm) [inline src/fontr/fontr/classifier.py:109]: I don't remember how we did manage to fix this, can we document it in this PR?

File: src/fontr/fontr/transforms.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import torch
+import random
+
+
+class Patch(torch.nn.Module):
+    """
+    Creates patches from images
+    """
+
+    def __init__(self, patch_size=10, patch_width=96, patch_height=96):
```
Comment (madziejm) [inline src/fontr/fontr/transforms.py:11]: nit: I would call this npatch/patch_count for clarity

File: src/fontr/fontr/transforms.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        self.patch_size = patch_size
```
Comment (madziejm) [inline src/fontr/fontr/transforms.py:12]: They sample "fifteen test patches, each of which comes with different aspect ratios and views, from one test image" in the original paper. I guess the solution we have can work as a substitute though.

File: src/fontr/fontr/transforms.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import torchvision.transforms as transforms
+import torch
+import random
+
+
+class Patch(torch.nn.Module):
```
Comment (madziejm) [inline src/fontr/fontr/transforms.py:6]: nit: I would call this "RandomPatches"

File: src/fontr/fontr/transforms.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        self.patch_height = patch_height
+
+    def __call__(self, image):
+        c, h, w = image.shape
+
+        patches = []
+
+        for _ in range(self.patch_size):
+            if w > self.patch_width:
+                left = random.randint(0, w - self.patch_width)
```
Comment (madziejm) [inline src/fontr/fontr/transforms.py:23]: hmm, this randomness actually makes inference non-determinant
let's leave this this way anyway; I see they don't care in the original paper

File: src/fontr/fontr/transforms.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            else:
+                up = 0
+                down = self.patch_height
+
+            after = image[0, up:down, left:right].unsqueeze(0)
+            patches.append(after)
+        return torch.stack(patches)
+
+
+class AddGaussianNoise(torch.nn.Module):
```
Comment (madziejm) [inline src/fontr/fontr/transforms.py:41]: How about [https://pytorch.org/vision/stable/generated/torchvision.transforms.GaussianBlur.html](https://pytorch.org/vision/stable/generated/torchvision.transforms.GaussianBlur.html)?

**Comment by madziejm** [comment]
Some notes:
1. The PR copies some code from https://github.com/madziejm/project-fontr/commit/9b754f766fe2e5352c6fc3ece5c81561d2d06365 that has not been merged
2. We still have some TODOs like the one to set the learning rate as in the original paper

**Review by madziejm — APPROVED** [review]
Ship it!
